x/magpie/internal/types/post.go

```diff
@@ -0,0 +1,105 @@
+package types
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// ---------------
+// --- Post id
+// ---------------
+
+// PostId represents a unique post id
+type PostId uint64
+
+func (id PostId) Valid() bool {
+	return id != 0
+}
+
+func (id PostId) Next() PostId {
+	return id + 1
+}
+
+func (id PostId) String() string {
+	return strconv.FormatUint(uint64(id), 10)
+}
+
+func ParsePostId(value string) (PostId, error) {
+	intVal, err := strconv.ParseUint(value, 10, 64)
+	if err != nil {
+		return PostId(0), err
+	}
+
+	return PostId(intVal), err
+}
+
+// ---------------
+// --- Post
+// ---------------
+
+// Post is a struct of a Magpie post
+type Post struct {
+	Id            PostId         `json:"id"`
```

Comment:
The `ID` is suggested by the golang linter. And this is how the modules in the `cosmos-sdk` is building. I prefer we follow the convention. Ex. `ProposalID` in `gov` module. https://github.com/cosmos/cosmos-sdk/blob/master/x/gov/types/proposal.go#L20